Cli/Command/CheckViolationsCommand.php

```diff
@@ -0,0 +1,150 @@
+            $diff = $gitRepository->getHeadCommit()->getDiff();
+        }
+
+        if (!$diff) {
+            $output->writeln('<info>No diff found.</info>');
+
+            return;
+        }
+
+        $changedFiles = array();
+        foreach ($diff->getFiles() as $file) {
+            if ($file->getNewName()) {
+                if (!$ignoreAssets && $this->isAsset($file->getNewName())) {
+                    continue;
+                }
+
+                $changedFiles[] = $file->getNewName();
+            }
+        }
+
+        $output->writeln(sprintf('<info>Found [%d] changed files</info>', count($changedFiles)));
+        $output->writeln(sprintf('<info>Get violations from latest SensioLabs insight analysis...</info>', count($changedFiles)));
+
+        $sensioInsightAnalysis = $api->getProject($input->getArgument('project-uuid'))->getLastAnalysis();
+        $analysisLinkPage = sprintf('%s/projects/%s/analyses/%d', $api::ENDPOINT, $projectUuid, $sensioInsightAnalysis->getNumber());
+
+        $output->writeln(
+            sprintf('<info>Last analysis number is [%d], link page <%s></info>', $sensioInsightAnalysis->getNumber(), $analysisLinkPage)
+        );
+
+        if (null === $sensioInsightAnalysis->getEndAt()) {
+            $output->writeln(
+                sprintf(
+                    '<error>The analysis number [%d], started at [%s] is not yet finished</error>',
+                    $sensioInsightAnalysis->getNumber(),
+                    $sensioInsightAnalysis->getBeginAt()->format('d D Y, H:i:s')
+                )
+            );
+
+            return 1;
+        }
+
+        if (!$sensioInsightAnalysis->getViolations()) {
+            $output->writeln(
+                sprintf('<info>No violations found in the last analysis [%d]</info>', $sensioInsightAnalysis->getNumber())
+            );
+
+            return;
+        }
+
+        $violationsByResources = array();
+        $globalViolations = array();
+        foreach ($sensioInsightAnalysis->getViolations() as $violation) {
+            if (!$violation->getResource()) {
+                $globalViolations[] = $violation->getMessage();
+                continue;
+            }
+
+            if (in_array($violation->getResource(), $changedFiles)) {
+                $violationsByResources[$violation->getResource()][] = $violation->getMessage();
+            }
+        }
+
+        if (count($violationsByResources)) {
+            foreach ($violationsByResources as $resource => $violations) {
+                $output->writeln(sprintf("\r\n<error>%s</error>", $resource));
+                foreach ($violations as $violationIndex => $violation) {
+                    $output->writeln(sprintf('<fg=yellow>    %d > %s</fg=yellow>', $violationIndex + 1, $violation));
+                }
+            }
+
+        } else {
+            $output->writeln(
+                sprintf('<info>No violations found in the last analysis [%d]</info>', $sensioInsightAnalysis->getNumber())
```

Comment:
this is still wrong if some global violations have been found, as you display them in case no violations have been found in the changed files.
It is also wrong in case violations have only been found in non-changed files (displaying a message is right, but the text being displayed is wrong)